src/streams.c

```diff
@@ -81,67 +83,121 @@ static Int READ_COMMAND(Obj *evalResult)
 
 
 /*
- * This function reads all code from the stream and returns either `fail` if the stream
- * cannot be opened, or executes all statements in the stream and returns a list of lists,
- * each entry of which reflects the result of the execution of one statement.
+ * This function reads all code from the stream and returns either `fail` if
+ * the stream cannot be opened, or executes all statements in the stream and
+ * returns a list of lists, each entry of which reflects the result of the
+ * execution of one statement.
  *
- * The results are returned as lists with at most three bound positions.
+ * The results are returned as lists of length three if outputFunc is
+ * false, or five, if outputFunc is a function.
  *
- * If the first entry is `true`, then the second entry is bound to the result of
- * the statement if there was one, and unbound otherwise, and the third entry
- * reflects whether the statement ended in a dual semicolon,
- * if the first entry is `false', an error occurred during the execution of a statement,
- * and no other positions of the list are bound.
+ * If the first entry is `true`, then the second entry is bound to the result
+ * of the statement if there was one, and unbound otherwise, and the third
+ * entry reflects whether the statement ended in a dual semicolon, if the
+ * first entry is `false', an error occurred during the execution of a
+ * statement, and no other positions of the list are bound. If outputFunc
```

Comment:
> ... and no other positions of the list are bound
This is (a) factually wrong (if `outputFunc` is given, then position 5 is always set), and (b) in the cases where it is not, it leads to a semi-broken plist being returned, one which has length 3, but indeed only the first position is set. In the latter case, I'd suggest to fix the length of the list; in the former case, the documentation or the code need to be adapted.